### PR TITLE
Part 2 of tutorial guide (part 1 yet to be written)

### PR DESCRIPTION
**Pull request by thomasmorgan**
## Description
Adds several new pages to the tutorials going through all the database tables in detail at a level suitable for a coding novice

## Motivation and Context
Will help people learn to use dallinger

### CODE REVIEW

File: docs/source/dummies_participant.rst [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+bad_data - Dallinger also allows experiments to check that a participants data is correct (the right number of nodes etc). If a participant fails this check their status will be set to bad_data. A few such cases are pretty much unavoidable - participants do strange and reckless things. But if you're getting a lot of these then there's probably a deeper issue. Unless instructed otherwise, Dallinger will automatically recruits a replacement participant.
+missing_notification - Occasionally the communication system between Dallinger and Mechanical Turk fails. This can cause serious problems, but Dallinger now checks for this. The most comon case is that a participant returns or abandons but Mechanical Turk never informs us. When this happens the relevant participant will be set to missing_notification.
+replaced - Another oddity with Mechanical Turk is that sometimes it recycles specific assignments if a participant returns it very shortly after accepting it. Again, Dallinger now tries to catch these instances and uses the status replaced to mark affected participants.
+
```
Comment (jessesnyder) [inline docs/source/dummies_participant.rst]: We now have `overrecruited`, also, to mark a participant who accepted a HIT which is already at capacity.

Comment (jessesnyder) [reply docs/source/dummies_participant.rst]: @thomasmorgan Maybe I'm wrong about that, and maybe this is why the build is erroring immediately on Travis. Any reason to no rebase onto master now?

Comment (thomasmorgan) [reply docs/source/dummies_participant.rst]: @jessesnyder I think its out of date, the last commit prior to opening the branch was f307f7ad9925841597e06de09dec27201dd3b4e4. Rebasing now should be fine - the branch only adds new files and does not edit any existing ones. Though the main reason not to do so is my lack of proficiency in these sorts of things. Can you handle the rebasing? Alternatively, if you give me some hints I can probably figure it out (e.g. do I need to undo the PR before rebasing?)

Comment (jessesnyder) [reply docs/source/dummies_participant.rst]: @thomasmorgan Sure, I'll try the rebase. I'll let you know when I've pushed the result.

Comment (jessesnyder) [reply docs/source/dummies_participant.rst]: Well, that turned out to be easy! Let's see how the Travis build goes.

**Comment by jessesnyder** [comment]
@thomasmorgan When I build the docs (`make -C docs html`) I get a lot of warnings. Many are because the "dummies" docs are not linked anywhere (for example):
```
/Users/jesses/py_dev/Dallinger3/docs/source/dummies_question.rst: WARNING: document isn't included in any toctree
```
or because there are placeholder links to docs not yet created:
```
/Users/jesses/py_dev/Dallinger3/docs/source/dummies_toc.rst:6: WARNING: toctree contains reference to nonexisting document 'dummies_why_dallinger'
```
In addition to these, there are also a lot of syntax warnings:
```
/Users/jesses/py_dev/Dallinger3/docs/source/dummies_node.rst:54: WARNING: Block quote ends without a blank line; unexpected unindent.
```
I think these should definitely be cleaned up prior to merge.

I'll turn my attention to the content now.

**Comment by vlall** [comment]
A few more warnings```
reading sources... [100%] dallinger_the_scientist                                                                                       
/Users/vlall/prog/dev/Dallinger/docs/source/gsoc.rst:85: WARNING: Bullet list ends without a blank line; unexpected unindent.
looking for now-outdated files... none found
pickling environment... done
checking consistency... /Users/vlall/prog/dev/Dallinger/docs/source/demos/iterated_drawing/index.rst: WARNING: document isn't included in any toctree
/Users/vlall/prog/dev/Dallinger/docs/source/demos_on_heroku.rst: WARNING: document isn't included in any toctree
/Users/vlall/prog/dev/Dallinger/docs/source/gsoc.rst: WARNING: document isn't included in any toctree
/Users/vlall/prog/dev/Dallinger/docs/source/learning_to_use_dallinger.rst: WARNING: document isn't included in any toctree
/Users/vlall/prog/dev/Dallinger/docs/source/required_experiment_files.rst: WARNING: document isn't included in any toctree
/Users/vlall/prog/dev/Dallinger/docs/source/vagrant_setup.rst: WARNING: document isn't included in any toctree
done
preparing documents... WARNING: unsupported theme option 'logo_name' given
done
writing output... [100%] writing_bots                                   
```

**Comment by thomasmorgan** [comment]
@vlall these are all issues with existing docs, not the new docs added by this PR

File: docs/source/dummies_node.rst [hunk 2]
```diff
@@ -0,0 +1,650 @@
...
+
+The next row creates the network_id column. It contains an integer (not a String). The next bit (`ForeignKey('network.id')`) might seem a bit mysterious, but again it can be solved by thinking about row/object duality. Recall that if you want to know details of a nodes network you can just do something like:
+::
+
+	node.network.creation_time
+
+well this ForeignKey is what let's you do that. It says that whatever you get by typing `node.network_id`, you should get the same thing by typing `node.network.id`. In fact it doesn't just say that you *should* get the same thing, it says you *will* get the same thing. This is just firmly cementing the relationship between a node and its network.
+
+The final bit (`index=True`) really doesn't need to trouble you. It's basically an instruction to the database to keep tabs on this column. In practice it slightly slows down row creation, but hugely speeds up searches across the table. If we didn't have this set to true experiments would steadily slow down as more participants took part in the experiment and the tables grew.
```
Comment (witekdev) [inline docs/source/dummies_node.rst]: "It's basically an instruction to the database to keep tabs on this column."
An unintentional pun here?

File: docs/source/dummies_node.rst [hunk 3]
```diff
@@ -0,0 +1,650 @@
...
+
+The final column is `details`. This serves a very similar function to the property columns discussed above, but is fancier and generally better. Chances are that down the line `details` will entirely replace the property columns and so this bit of the guide will need to be rewritten.
+
+Node objects
+------------
+
+So far we've covered Node's from the table view, but remember that all Dallinger classes have table/object duality and in general the object side of things is far more useful. So what are the extra features of Nodes if we treat them as objects? (In a good way.) Let's return to the Node class in models.py and look immediately below where the columns were created. The first function is `__init__`:
```
Comment (witekdev) [inline docs/source/dummies_node.rst]: "(In a good way.)"
I am not following this part of the comment? :thinking:

Comment (witekdev) [reply docs/source/dummies_node.rst]: Is this a reference to treating Nodes as if not having a soul or something? :laughing:

Comment (thomasmorgan) [reply docs/source/dummies_node.rst]: I'm not sure either - this is what happens when I binge write documentation! Me and Jesse have been tweaking the docs on the branch `jesse-edits` feel free to remove it there (Jesse, who did the dummies_node tweaks may have already done so).

Comment (witekdev) [reply docs/source/dummies_node.rst]: yeah, I'm working my way through that branch and making tweaks. Much ground to cover :muscle:

File: docs/source/dummies_transmission.rst [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+
+1. A makes the info
+2. A transmits it to B
+3. B receives it, and makes a new info with the same contents (perhaps linking the new info to the old one via a transformation)
+4. B transmits the new info to C
+
+This is why, back in the previous page, I said the London Underground analogy isn't brilliant for Infos. In the London Underground each train (which is sort of like an Info, sort of) can easily be sent along a very long journey without anything serious going wrong. Moreover, trains don't have an "origin" station where they were made. Instead the Underground network contains a finite number of trains that are continually suffled around the network and were made somewhere quite different. In Dallinger things are different. Infos are continually being made by Nodes all the time, and while those Nodes can sent ther Infos to other Nodes they have a connection with, those Nodes cannot send them any further. Instead they have to duplicate them, making a new Info, and transmit the duplicant.
```
Comment (witekdev) [inline docs/source/dummies_transmission.rst]: https://shop.tcgplayer.com/magic/mirrodin/duplicant 
:smile:

**Pull request closed without merging**